File: 1878-check-if-array-is-sorted-and-rotated/check-if-array-is-sorted-and-rotated.rs
impl Solution {
    pub fn check(nums: Vec<i32>) -> bool {
        let mut is_down = false;
        for i in 0..nums.len() {
            let prev = if i == 0 { nums.len() - 1 } else { i - 1 };
            if nums[prev] > nums[i] {
                if is_down {
                    return false;
                } 
                is_down = true;
            }
        }
        true
    }
}

File: 1878-check-if-array-is-sorted-and-rotated/check-if-array-is-sorted-and-rotated.ts
function check(nums: number[]): boolean {
    let rotated = false;
    for(let i = 0; i < nums.length; i++) {
        const prev = i > 0 ? i - 1: nums.length - 1;
        if(nums[i] < nums[prev]) {
            if(rotated) return false;
            rotated = true;
        }
    }
    return true;
};